frontend/projects/abtesting/src/app/features/dashboard/home/components/experiment-design/experiment-design.component.ts

```diff
@@ -243,11 +243,20 @@ export class ExperimentDesignComponent implements OnInit, OnChanges, OnDestroy {
   }
 
   validateConditionCodes(conditions: ExperimentCondition[]) {
+    let defaultKeywordError = false;
     const conditionCodes = conditions.map(condition => condition.conditionCode);
+    conditionCodes.forEach(conditioncode => {
+      if (conditioncode.toUpperCase() === this.translate.instant('home.new-experiment.design.condition.invalid.text')) {
```

Comment:
It seems two different validations are going on in this function. Can we have a separate method for the validation for default keyword? may be hasConditionCodeDefault()?

Another thing is instead of translating the text in each iteration, take this out from the forEach loop.

Reply:
ok

Reply:
Done now